File: 2374_Node_With_Highest_Edge_Score.cpp
class Solution {
public:
    int edgeScore(vector<int>& edges) 
    {
        int n = edges.size();
        vector<int64_t> weights(n);
        for (int i = 0; i < n; ++i)
            weights[edges[i]] += i;

        int64_t mxw = INT_MIN;
        int ret = -1;
        
        for (int i = 0; i < n; ++i)
        {
            if (weights[i] > mxw)
            {
                mxw = weights[i];
                ret = i;
            }
        }

        return ret;
    }
};

File: py/2374_Node_With_Highest_Edge_Score.py
class Solution(object):
    def edgeScore(self, edges):
        """
        :type edges: List[int]
        :rtype: int
        """
        
        n = len(edges)
        weights = [0] * n

        for i in range(0, n):
            weights[edges[i]] += i

        mxw = float('-inf')
        ret = -1

        for i in range(0, n):
            if (weights[i] > mxw):
                mxw = weights[i]
                ret = i

        return ret
